### PR TITLE
Hide sensitive AST/config data unless --debug-config argument passed

### PR DESCRIPTION
**Pull request by andrewvc**
We now hide this because displaying it is dangerous. Generated code
and other AST data may contain plaintext copies of 'password' type fields.
Users can force this to appear with the --debug-config flag.

Fixes https://github.com/elastic/logstash/issues/4964

### CODE REVIEW

**Comment by ph** [comment]
Something seems weird with this PR, when I change my Gemfile to point the `logstash-core` and `logstash-core-event` to the local directory and I bootstrap the environment the logstash executable dont do anything. It stop without giving warnings or errors.

```
 bin/logstash -e 'input { generator {}}'
```

**Comment by ph** [comment]
@andrewvc do you have the same issue locally?

**Comment by andrewvc** [comment]
@ph I do have the same issue. Something is borked in 2.2

**Comment by ph** [comment]
@andrewvc I have clone the 2.2 branch locally and the `bin/logstash -e 'input { generator {}}'` works successfully so I think the problem is coming from that PR, so It should be a problem with the backport.

**Comment by andrewvc** [comment]
I just rebased this off origin/2.2 and the tests pass locally. Let's hope they do no jenkins!

**Comment by purbon** [comment]
please jenkins, test this.

**Comment by purbon** [comment]
@andrewvc sorry to tell you is still failing.

**Comment by ph** [comment]
Finished in 13.93 seconds (files took 4.49 seconds to load)
1295 examples, 0 failures

Randomized with seed 1924

**Comment by ph** [comment]
LGTM

File: logstash-core/spec/logstash/runner_spec.rb [hunk 1]
```diff
@@ -40,7 +40,7 @@ def run(args); end
 
   describe "pipeline settings" do
     let(:pipeline_string) { "input { stdin {} } output { stdout {} }" }
-    let(:base_pipeline_settings) { { :pipeline_id => "base" } }
+    let(:base_pipeline_settings) { { :pipeline_id => "base", :debug_config => false } }
```
Comment (talevy) [inline logstash-core/spec/logstash/runner_spec.rb]: why is this required here?

**Comment by talevy** [comment]
@andrewvc left a small comment, otherwise LGTM

File: logstash-core/spec/logstash/runner_spec.rb (hunk 1 above)
Comment (andrewvc) [inline logstash-core/spec/logstash/runner_spec.rb]: Just there to indicate the default value the Agent passes in is false!

Comment (talevy) [inline logstash-core/spec/logstash/runner_spec.rb]: ah, otherwise behavior in this isolated test will be different?

Comment (andrewvc) [inline logstash-core/spec/logstash/runner_spec.rb]: No, it's there more as a semantic guide.
